cluster-api-1.9.yaml

```diff
@@ -0,0 +1,136 @@
+    items:
+      kubeadm-bootstrap-controller: bootstrap/kubeadm
+      kubeadm-controlplane-controller: controlplane/kubeadm
+      clusterctl: cmd/clusterctl
+
+subpackages:
+  - range: backends-go-builds
+    name: ${{package.name}}-${{range.key}}
+    description: cluster-api core components
+    dependencies:
+      provides:
+        - cluster-api-${{range.key}}=${{package.full-version}}
+    pipeline:
+      - uses: go/build
+        with:
+          output: ${{range.key}}
+          packages: ./${{range.value}}
+          ldflags: |
+            -X sigs.k8s.io/cluster-api/version.gitVersion=${{package.version}} \
+            -X sigs.k8s.io/cluster-api/version.gitCommit=${GIT_COMMIT} \
+            -X sigs.k8s.io/cluster-api/version.gitTreeState=$(shell git diff --quiet >/dev/null 2>&1 || echo "dirty" || echo "clean") \
+            -X sigs.k8s.io/cluster-api/version.buildDate=$(date -u +'%Y-%m-%dT%H:%M:%SZ')
+
+  - range: backends-go-builds
+    name: ${{package.name}}-${{range.key}}-compat
+    description: cluster-api core ${{range.key}} compat packages
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}/
+          ln -s /usr/bin/${{range.key}} ${{targets.contextdir}}/${{range.key}}
+
+  - name: ${{package.name}}-controller
+    description: cluster-api core controller
+    dependencies:
+      provides:
+        - cluster-api-controller=${{package.full-version}}
+    pipeline:
+      - uses: go/build
+        with:
+          output: manager
+          packages: .
+          ldflags: |
+            -X sigs.k8s.io/cluster-api/version.gitVersion=${{package.version}} \
+            -X sigs.k8s.io/cluster-api/version.gitCommit=${GIT_COMMIT} \
+            -X sigs.k8s.io/cluster-api/version.gitTreeState=$(shell git diff --quiet >/dev/null 2>&1 || echo "dirty" || echo "clean") \
+            -X sigs.k8s.io/cluster-api/version.buildDate=$(date -u +'%Y-%m-%dT%H:%M:%SZ')
+
+  - name: ${{package.name}}-controller-compat
+    description: cluster-api core controller compat package
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}/
+          ln -s /usr/bin/manager ${{targets.contextdir}}/manager
+
+test:
+  environment:
+    contents:
+      packages:
+        - ${{package.name}}-controller
+        - ${{package.name}}-kubeadm-bootstrap-controller
+        - ${{package.name}}-kubeadm-controlplane-controller
+        - ${{package.name}}-clusterctl
+        - ${{package.name}}-kubeadm-bootstrap-controller-compat
+        - ${{package.name}}-kubeadm-controlplane-controller-compat
+        - ${{package.name}}-controller-compat
+        - ${{package.name}}-clusterctl-compat
+  pipeline:
+    - uses: git-checkout
+      with:
+        expected-commit: 813e90d8daef60b715764ce4729b40d82f811cc1
+        repository: https://github.com/kubernetes-sigs/cluster-api
+        tag: v${{package.version}}
+    - uses: test/kwok/cluster
+    - runs: |
+        kwokctl scale node --replicas 3
```

Comment:
Should we drop a comment why we need to scale the Nodes?

Reply:
Not needed, removed